src/databricks/labs/ucx/workspace_access/groups.py

```diff
@@ -165,11 +165,22 @@ def prepare_groups_in_environment(self):
             logger.info("Using the provided group listing")
 
             for g in group_names:
-                assert g not in self.SYSTEM_GROUPS, f"Cannot migrate system group {g}"
-                assert self._get_group(g, "workspace"), f"Group {g} not found on the workspace level"
-                assert self._get_group(g, "account"), f"Group {g} not found on the account level"
-
-        if not group_names:
+                if g in self.SYSTEM_GROUPS:
+                    logger.info(f"Cannot migrate system group {g}. {g} will be skipped.")
+                    group_names.remove(g)
```

Comment:
don't modify the list while iterating, as it's one of the most common sources for bugs. e.g. don't use `group_names.remove`, it doesn't give deterministic results:

```
>>> x = [1,2,3,4,5]
>>> for i in x:
...   if i > 3:
...     x.remove(i)
... 
>>> x
[1, 2, 3, 5] <=== 5 should have been removed

>>> for i in x: x.remove(i)
... 
>>> x
[2, 4] <==== should have been empty
```

instead, create a copy of the list, but with invalid elements removed.

Reply:
Done